synapse/lib/storm.py

```diff
@@ -950,12 +949,12 @@ def _stormOperJoinTags(self, query, oper):
         tags = {tag for node in nodes for tag in s_tufo.tags(node, leaf=True)}
 
         if not forms:
-            forms = core.getModelDict().get('forms')
+            forms = set(core.getModelDict().get('forms'))
```

Comment:
seems like DataModel needs a getFormNames() API that just returns a list ( i can see other bits of code wrangling this concept too )